deploy/docker/notebook/base/base.yaml

```diff
@@ -421,26 +425,110 @@ addfiles:
-          # Configure file post-save hook
-          def script_post_save(model, os_path, contents_manager, **kwargs):
-              work_path = '/home/jovyan/work/'
-              shared_path = '/home/jovyan/shared/notebooks'
-
-              if not (is_inside_path(os_path, work_path) or is_inside_path(os_path, shared_path)):
-                    return
-
-              logging.getLogger().info(f"Saved notebook at {os_path}")
-              try:
-                    r = requests.post("https://polus-notebooks-hub-backend.ci.aws.labshare.org/file-data/new-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
-                    if r.status_code == 200:
-                        logging.getLogger().info(f"Registered {os_path} with Notebooks Hub")
-                    else:
-                        logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub. Return code: {r.status_code}")
-              except Exception as e:
-                    logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub", exc_info=e)
-              
+          # Configure file post-save, post-rename, and post-delete hooks
           
-          c.FileContentsManager.post_save_hook = script_post_save
\ No newline at end of file
+					#### Hook Class ####
+
+					class HookFileContentsManager(FileContentsManager):
+							"""Local spawner that runs single-user servers as the same user as the Hub itself.
+							Overrides user-specific env setup with no-ops.
+							"""
+							
+							post_delete_hook = Any(None, config=True,
+									help="""Python callable thereof
+									to be called after delete a notebook.
+									"""
+							)
+
+							post_save_hook = Any(None, config=True,
+									help="""Python callable thereof
+									to be called after delete a notebook.
+									"""
+							)
+
+							post_rename_hook = Any(None, config=True,
+									help="""Python callable thereof
+									to be called after update a notebook.
+									"""
+							)
+
+							def delete(self, os_path):
+									returned = super(HookFileContentsManager, self).delete_file(os_path)
+									if self.post_delete_hook:
+											self.post_delete_hook(os_path=os_path, contents_manager=self)
+									return returned
+
+							def save(self, model, os_path):
+									returned = super(HookFileContentsManager, self).save(model, path=os_path)
+									if self.post_save_hook:
+											self.post_save_hook(model=model, os_path=os_path, contents_manager=self)
+									return returned
+
+							def rename(self, os_old_path, os_new_path):
+									returned = super(HookFileContentsManager, self).rename_file(os_old_path, os_new_path)
+									if self.post_rename_hook:
+											self.post_rename_hook(os_old_path=os_old_path, os_new_path=os_new_path, contents_manager=self)
+									return returned
+
+					#### Set contents manager class ####
+
+					c.NotebookApp.contents_manager_class = HookFileContentsManager
```

Comment:
Could you only set the logging class conditionally, if `POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED` is set to `'True'`? This way, there is a switch to enable/disable communication with the Notebooks Hub API. The reason for having this is to 1) enable local Docker image debugging outside the cluster; 2) being able to turn off the hooks if Notebooks Hub goes down (if API goes down, routine actions such as saving/renaming/deleting of notebook files will stop working)